src/screens/login/components/OidcOnboardingView.tsx

```diff
@@ -0,0 +1,102 @@
+import Minilog from '@cozy/minilog'
+import React, { useEffect, useState } from 'react'
+import { KeyboardAvoidingView, Platform, StyleSheet, View } from 'react-native'
+
+import { SupervisedWebView } from '/components/webviews/SupervisedWebView'
+
+import { parseOidcOnboardingFinishedUrl } from './functions/oidc'
+
+import { NetService } from '/libs/services/NetService'
+import { routes } from '/constants/routes'
+
+const log = Minilog('OidcOnboardingView')
+
+interface ClouderyWebViewProps {
+  code: string
+  onboardUrl: string
+  startOidcOAuth: (fqdn: string, code: string) => void
+}
+
+/**
+ * Display a WebView with specidied 'onboardUrl' and call startOidcOauth when
+ * navigation to OIDCInstanceCreationResult url occurs
+ */
+export const OidcOnboardingView = ({
+  code,
+  onboardUrl,
+  startOidcOAuth
+}: ClouderyWebViewProps): JSX.Element => {
+  const [loading, setLoading] = useState(true)
+  const [displayOverlay, setDisplayOverlay] = useState(true)
+
+  const handleNavigation = (request: WebViewNavigation): boolean => {
+    const createdInstance = parseOidcOnboardingFinishedUrl(request)
+    if (createdInstance) {
+      startOidcOAuth(createdInstance.fqdn, code)
+      return false
+    }
+
+    return true
+  }
+
+  useEffect(() => {
+    // Wait 1ms before hiding overlay to prevent screen flickering
+    if (!loading) {
+      setTimeout(() => {
+        setDisplayOverlay(false)
+      }, 1)
+    }
+  }, [loading])
```

Comment:
🤔 Between which screens? Can't the first screen send an event or else? 1ms is totally arbitrary, what about old fashion device?

Reply:
I'm not sure what exactly happens, I just copied the code from ClouderyView based on this PR: https://github.com/cozy/cozy-flagship-app/pull/308/commits/241a428d78fc0e887a47875b7d7a17a50be3637f

What I can tell you:
- `Can't the first screen send an event or else?` -> this is already what is happening. But Pierre had to add this timeout mechanism to prevent a flash.
- `1ms is totally arbitrary` -> in React and Angular worlds, when you see a timeout with 1ms you can assume that the timeout is here just for "delaying" the effect to the next render loop.

Reply:
> 1ms is totally arbitrary -> in React and Angular worlds, when you see a timeout with 1ms you can assume that the timeout is here just for "delaying" the effect to the next render loop.

If the goal is to wait the next tick, then setTimeout(0) do the work. It's clearer than 1. No?